maro/simulator/scenarios/vm_scheduling/region.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List
+
+from maro.backends.frame import NodeAttribute, NodeBase, node
+
+
+@node("regions")
+class Region(NodeBase):
+    """Region node definition in frame."""
+    id = NodeAttribute("i2")
+
+    total_machine_num = NodeAttribute("i")
```

Comment:
how about implementing the `empyt_machine_num` as an attribute that traverses the `self._xx_list` and returns the sum of their empty_machine_num?

Reply:
But the `self._machines` or `self._racks` or .. is maintained by BE, how can each class get "empty" machines? The `self._xx_list` only store ids.

Reply:
You're right